src/ImportDefinitionsBundle/Importer/Importer.php

```diff
@@ -153,12 +153,12 @@ public function doImport(DefinitionInterface $definition, $params)
         /** @var ImportDataSetInterface|array $data */
         $data = $this->getData($definition, $params);
 
-        if (\count($data) > 0) {
+        if ((is_array($data) || $data instanceof \Countable) && \count($data) > 0) {
```

Comment:
Only count if actually https://www.php.net/manual/en/function.is-countable.php

(cannot use this until we're 7.3+)

Reply:
👍

Reply:
How does the progress bar work if you don't give it a count? How does that look?

Reply:
Yes, that might be a problem: currently there is none.

Reply:
We can maybe implement a countable iterator?

Reply:
this makes sense _if_ you have some way of getting the count without unrolling the iterator and losing the lazy benefits

Reply:
Yes, we rely too heavily on knowing the count for various functionality bits so I've dropped the generator approach.